openhands/controller/agent_controller.py

```diff
@@ -249,6 +250,12 @@ async def _handle_observation(self, observation: Observation):
         elif isinstance(observation, ErrorObservation):
             if self.state.agent_state == AgentState.ERROR:
                 self.state.metrics.merge(self.state.local_metrics)
+        elif isinstance(observation, FatalErrorObservation):
+            await self.report_error(
+                'There was a fatal error during agent execution: ' + str(observation)
+            )
+            await self.set_agent_state_to(AgentState.ERROR)
+            self.state.metrics.merge(self.state.local_metrics)
```

Comment:
This seems the same with how ErrorObservation is treated? `report_error` will create an ErrorObs

Reply:
But i think set state to ERROR will terminate the session?

Reply:
We probably don't need to report_error if that's the case..